### PR TITLE
AssetMeta - Change documents ordering

### LINKED ISSUE #157: Change ordering of Document categories
**Problem today**
The DocumentTable component sorts document categories seemingly arbitrarily. As a user, it is hard for me quickly to find the documents I'm looking for. E.g., when I investigate pumps, I care about the design data for pumps which lives in one of the pump's Data Sheets.

**Proposed solution**
- Always show "Logic Diagrams" and "P&IDs" on top of the list. These are by far the most used documents. 
- List the other document categories in ascending alphabetical order. 
- In order to explain the ordering, perhaps somehow visually distinguish between Logic Diagrams / P&IDs on top and the rest of the document categories. If not, it might just look like the categories are weirdly out of order again.

### PR DESCRIPTION
**Pull request by xitman78**
Show prioritized categories on top of the list.
Make categories "P&ID" and "Logic Diagrams" to be prioritized by default.
Show delimiter (thick line) between prioritized categories and regular categories.
Sort regular categories in alphabetical order by default.
Added prop for custom categories sort.

Fixes #157

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-171.gearboxjs.preview.cogniteapp.com

File: src/components/AssetMeta/components/DocumentTable.tsx [hunk 1]
```diff
@@ -65,19 +67,22 @@ export class DocumentTable extends React.PureComponent<
...
     );
-    const categoryByPriority = getCategoryByPriority(
-      documentsByCategory,
-      categoryPriorityList
-    );
+    const {
+      categories,
+      prioritizedCount,
+    }: { categories: string[]; prioritizedCount: number } = customCategorySort
+      ? getCategoryByCustomSort(documentsByCategory, customCategorySort)
```
Comment (mrtraser) [inline src/components/AssetMeta/components/DocumentTable.tsx]: Could we have prioritized array always in a game? Like:
1. Apply prioritized rules from `categoryPriorityList` (default or provided)
2. Then sort others by default/custom sort function

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-171.gearboxjs.preview.cogniteapp.com

File: src/components/AssetMeta/components/DocumentTable.tsx (hunk 1 above)
Comment (xitman78) [reply src/components/AssetMeta/components/DocumentTable.tsx]: done

**Pull request merged**